### ISSUE TITLE
Dataflow edge cardinality counters

### ISSUE THREAD

**Issue by MingweiSamuel** [hydro-project/hydro#1269]
can extract at end of `run()` call

1. Should this be code generated by hydroflow plus? Probably no
2. Should this be a hydroflow operator? Basically `identity()` with `count++`
    - will pollute hydroflow graphs
4. or should we integrate it into the hydroflow state/context core more directly. as a new feature for #context

**Comment by MingweiSamuel** [comment]
Maybe a hybrid of 2 and 3

**Comment by MingweiSamuel** [comment]
(Is option 1 already implemented?)

**Comment by davidchuyaya** [comment]
Yup, that's this file: https://github.com/hydro-project/hydroflow/blob/main/hydroflow_plus/src/profiler.rs

**Comment by MingweiSamuel** [comment]
29a263fb564c5ce4bc495ea4e9d20b8b2621b645

**Comment by shadaj** [comment]
We need a version of this baked into DFIR, since we can't use `RuntimeData` anymore in Hydro code (and also using `RefCell` for counters was not good for performance).

What we need: annotate which operators we want counters for, provide global configuration for how frequently to print out counter data (stdout is fine for now, we can use operator IDs like for perf).

**Comment by MingweiSamuel** [comment]
Moved to #1729
